### PR TITLE
[somfytahoma] Fixed bug causing Gateway and Actiongroup never get ONLINE

### PR DESCRIPTION
**Pull request by octa22**
This PR fixes bug causing Gateway and Actiongroups never get ONLINE introduced by #10300

might relate to #10346

### CODE REVIEW

**Comment by lolodomo** [comment]
I am so sorry to have broken something. 
But I would like to propose to you a proper fix.

**Comment by octa22** [comment]
> But I would like to propose to you a proper fix.

OK, feel free to come up with a proper fix. But keep in mind that Actiongroup things have no availability status (as well as non IO homecontrol devices) and Gateway's status is returned in a completly different way.

**Comment by lolodomo** [comment]
> OK, feel free to come up with a proper fix. But keep in mind that Actiongroup things have no availability status (as well as non IO homecontrol devices) and Gateway's status is returned in a completly different way.

You last version is almost ok, the only change you should consider is to override the method `initializeThing` rather than the method `initialize`. Doing that, this will handle properly any bridge status change.

**Comment by octa22** [comment]
> You last version is almost ok, the only change you should consider is to override the method `initializeThing` rather than the method `initialize`. Doing that, this will handle properly any bridge status change.

Good point, I have overrided the `initializeThing` method instead.

**Comment by lolodomo** [comment]
I think it is OK now, I have just a little doubt if we should check if the bridge handler is not null.

**Review by lolodomo — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
@openhab/add-ons-maintainers : please merge this PR.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java [hunk 1]
```diff
@@ -34,8 +34,16 @@ public SomfyTahomaActionGroupHandler(Thing thing) {
...
+    public void initializeThing(@Nullable ThingStatus bridgeStatus) {
+        if (bridgeStatus != null) {
+            if (bridgeStatus == ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
+        }
```
Comment (fwolter) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java:46]: If this Thing should be online if and only if the bridge is online. You can simply set it to ONLINE in `initialize()`. The framework takes care of the state, then.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java [hunk 2]
```diff
@@ -15,9 +15,9 @@
 import static org.openhab.binding.somfytahoma.internal.SomfyTahomaBindingConstants.EXECUTE_ACTION;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
 import org.openhab.core.library.types.OnOffType;
-import org.openhab.core.thing.ChannelUID;
-import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.*;
```
Comment (wborn) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java:20]: Please don't use star imports. Update your IDE settings.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java [hunk 3]
```diff
@@ -12,14 +12,14 @@
...
 import static org.openhab.core.thing.Thing.PROPERTY_FIRMWARE_VERSION;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
 import org.openhab.binding.somfytahoma.internal.model.SomfyTahomaStatus;
 import org.openhab.core.library.types.StringType;
-import org.openhab.core.thing.Channel;
-import org.openhab.core.thing.Thing;
-import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.*;
```
Comment (wborn) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java:22]: Please don't use star imports. Update your IDE settings.

**Review by wborn — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java:46]: This method overrides a more complex method used for all things defined in the parent class `SomfyTahomaBaseThingHandler`.

Comment (fwolter) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java:46]: I see. `bridgeStatusChanged()` is overriden.

There's a HTTP request done in `initialize()`. Would be awesome if you could fix this in another PR!

Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java:46]: Yes, I will do it in another PR.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java (hunk 2 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java:20]: Fixed

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java (hunk 3 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java]: Fixes

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java (hunk 1 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java:46]: Guys, so what's wrong with this code, should I change something in this PR?

Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java:46]: Nothing to change un m'y opinion.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java (hunk 2 above)
Comment (wborn) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java:20]: It's still there.

Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaActionGroupHandler.java]: you are right. now it is fixed

**Review by wborn — APPROVED** [review]
Thank you all!

**Pull request merged**